pages/unitCard.tsx

```diff
@@ -1,40 +1,98 @@
 import { NextPage } from "next";
-import { getWeaponData, WeaponData } from "../src/unitStats/WeaponLib";
-import { DpsChart } from "../components/unitStats/DpsChart";
+import { DpsChart } from "../components/unitStats/dpsChart";
+import { EbpsType, getEbpsStats, setEbpsStats } from "../src/unitStats/mappingEbps";
+import { getSbpsStats, SbpsType, setSbpsStats } from "../src/unitStats/mappingSbps";
+import {
+  getWeaponStats,
+  setWeaponStats,
+  WeaponStats,
+  WeaponType,
+} from "../src/unitStats/mappingWeapon";
+import {
+  getUpgradesStats,
+  setUpgradesStats,
+  UpgradesType,
+} from "../src/unitStats/mappingUpgrades";
+
+type Locstring = {
+  id: number;
+  value: string;
+};
+
+let unitStatsLocString: Locstring[];
+
+export const resolveLocstring = (locstring: any) => {
+  const numId = parseInt(locstring?.locstring?.value);
+
+  let result = unitStatsLocString?.find((entry) => entry.id == numId)?.value;
+  if (!result) result = "No text found.";
+
+  return result;
+};
 
 interface UnitCardProps {
-  weaponData: WeaponData[];
-  squadData: any;
+  weaponData: WeaponType[];
+  spbsData: SbpsType[];
+  epbsData: EbpsType[];
+  upgradesData: UpgradesType[];
   generalInfo: any;
   properties: any;
 }
 
 // Parameter in Curly brackets is destructuring for
 // accessing attributes of Props Structure directly
-const UnitPage: NextPage<UnitCardProps> = ({ weaponData }) => {
-  //const [searchValue, onSearchChange] = useState('');
+const UnitPage: NextPage<UnitCardProps> = ({ weaponData, spbsData, epbsData, upgradesData }) => {
+  if (!WeaponStats)
+    // Save data again in global varible for clientMode
+    setWeaponStats(weaponData);
+
+  if (!epbsData) setEbpsStats(epbsData);
+
+  if (!upgradesData) setUpgradesStats(upgradesData);
+
+  if (!spbsData) setSbpsStats(spbsData);
 
-  //<div style={{"height": "800px"}}>
   return (
     <div>
-      <DpsChart searchData={weaponData}></DpsChart>
+      <DpsChart></DpsChart>
     </div>
   );
 };
 
 export const getStaticProps = async () => {
-  // relative path are not supported as far as I understood
-  //const myReq = await fetch("http://localhost:3000/game_stats/weapons.json");
-  const myReq = await fetch(
-    "https://raw.githubusercontent.com/cohstats/coh3-data/xml-data/scripts/xml-to-json/exported/weapon.json",
-  );
+  //const weapon = await myReqWeapon.json();
+  // map Data at built time
+  const weaponData = await getWeaponStats();
+
+  // map Data at built time
+  const ebpsData = await getEbpsStats();
+
+  // map Data at built time
+  const sbpsData = await getSbpsStats();
+
+  // map Data at built time
+  const upgradesData = await getUpgradesStats();
```

Comment:
In my case, the `sbpsData` contains the extension with `template_reference.value` as `sbpextensions\\squad_upgrade_ext`. If I wanted to store the list of upgrades of an entity (unit / building), I have to store the `instance_reference` to that upgrade.

In order to properly map it, should we do the mapping after obtaining bpth the `sbpsData` and `upgradesData` in another step OR should we change the order of resolving such JSON files (upgrades first, then sbps) to instantly add the required upgrade data into the resolved `sbps` object?

**First option**
```json5
# sbpsData
{
  "l6_40_ak": {
    "icon_name": "L6/40 Tank",
    "upgrades": [
      "upgrade\\german\\vehicle\\spotting_scope_l6_40_ger",
      "upgrade\\afrika_korps\\vehicle\\flamethrower_l640_ak"
    ]
  }
}
```

**Second option**
```json5
# sbpsData
{
  "l6_40_ak": {
    "icon_name": "L6/40 Tank",
    "upgrades": [
      { "icon_name": "common\\abilities\\scan_kettenkrad_ger", "screen_name": "Spotting Scopes" },
      { "icon_name": "races\\afrika_corps\\upgrades\\lanciaflamme", "screen_name": "Lanciaflame L6"}
    ]
  }
}
```

Reply:
I would save the the extracted attribute data like in the first option and write a quick method that resolves the dependencies when prepearing the data for the components. This way we have a clear cut between the mapping variables in the cache and no redundancies. There are examples where a complete ebps is referenced but only one or two values are required.